inst/pages/98_exercises.qmd

```diff
@@ -1158,62 +1155,145 @@ See the OMA book for examples.
 **Extra** Cool Exercise : Identify Patterns and Clusters
 To make the exercise more engaging, let's do an exploratory data analysis task:
 
-#### 1. Cluster Analysis on PCoA/PCA Results:
+### 1: k-means Clustering on Reduced Dimensions and Visualization
 
-i. Perform clustering on the ordination results (e.g., k-means clustering).
-ii. Visualize the clusters on the ordination plots.
+- Performs k-means clustering on reduced dimensional data obtained from PCoA.
+- Visualizes the clusters in a scatter plot of the reduced dimensions.
 
 ```{r}
-# Setting seed for reproducibility
-set.seed(123)
-
-# Perform k-means clustering on the reduced dimensions of pcoa_bray
-pcoa_bray_clusters <- kmeans(reducedDim(pcoa_bray), centers = 3)$cluster
-
-# Plot the reduced dimensions of pcoa_bray with clusters as colors
-plotReducedDim(pcoa_bray, colour_by = pcoa_bray_clusters)
-
-# Perform k-means clustering on the reduced dimensions of pca_aitchison
-pca_aitchison_clusters <- kmeans(reducedDim(pca_aitchison), centers = 3)$cluster
+#| label: heatmap-visualization-ex3
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
 
-# Plot the reduced dimensions of pca_aitchison with clusters as colors
-plotReducedDim(pca_aitchison, colour_by = pca_aitchison_clusters)
+# Load necessary libraries
+library(mia)
+library(SummarizedExperiment)
+library(vegan)
+library(ggplot2)
+library(cluster)
+
+# Generate a sample SummarizedExperiment object
+data(GlobalPatterns)
+se <- SummarizedExperiment(assays = list(counts = assay(GlobalPatterns)),
+                           colData = colData(GlobalPatterns),
+                           rowData = rowData(GlobalPatterns))
+
+# Perform PCoA on the abundance data
+distance_matrix <- vegdist(t(assay(se)), method = "bray")
+pcoa_results <- cmdscale(distance_matrix, k = 2)
+pcoa_data <- data.frame(pcoa_results)
+colnames(pcoa_data) <- c("PCoA1", "PCoA2")
+
+# Perform k-means clustering (k = 3)
+set.seed(123)
+kmeans_result <- kmeans(pcoa_data, centers = 3)
+pcoa_data$cluster <- as.factor(kmeans_result$cluster)
```

Comment:
Instead use bluster package

Reply:
Changes made in this commit : 

1. In heatmap visualization ex-1:
- `scater::runMDS` is used to perform MDS on the abundance data.
- `bluster` package is used for hierarchical clustering and k-means clustering.
- `ComplexHeatmap` < – > “pheatmap”

2. In heatmap visualization ex-2:
- Changed to CLR+Euclidean 

3. In heatmap visualization ex-3:
- Assumes that the cluster information has already been added to `colData` and does not recalculate distances and clusters.

Just a question, Is there a specific reason we are using complexheatmap

Reply:
pheatmap and ComplexHeatmap are convenient tools to plot heatmaps. They are easier to use than ggplot2 (for making nicer heatmaps). ComplexHeatmap is the next iteration of pheatmap, it can do the same as pheatmap and more. In fact, it has ComplexHeatmap::pheatmap function that creates pheatmap-like plots.

sechm is a package to plot assay of SummarizedExperiment. It creates ComplexHeatmap object. We should use sechm whenever possible since it makes the examples simpler (user can always use ComplexHeatmap directly if wants).

Reply:
Okay I got what you mean now, thank you :)